### PR TITLE
feat: add support-bundle-node-collection-timeout config

### PR DESCRIPTION
**Pull request by Yu-Jack**
#### PR Checklist
- Is this a multi-tenancy feature/bug?
    - [ ] Yes, the relevant RBAC changes are at:
- Do we need to backport changes to the [old Rancher UI](https://github.com/rancher/u), such as RKE1?
    - [ ] Yes, the relevant PR is at:
- Are backend engineers aware of UI changes?
    - [ ] Yes, the backend owner is:

Fixes https://github.com/harvester/harvester/issues/1646

### Occurred changes and/or fixed issues

Add new `suppport-bundle-node-collection-timeout` setting.

### Screenshot/Video

image

### CODE REVIEW

File: pkg/harvester/config/settings.js [hunk 1]
```diff
@@ -1,35 +1,36 @@
...
+  UI_PL:                                  'ui-pl',
+  HTTP_PROXY:                             'http-proxy',
+  ADDITIONAL_CA:                          'additional-ca',
+  OVERCOMMIT_CONFIG:                      'overcommit-config',
+  CLUSTER_REGISTRATION_URL:               'cluster-registration-url',
+  DEFAULT_STORAGE_CLASS:                  'default-storage-class',
+  SUPPORT_BUNDLE_TIMEOUT:                 'support-bundle-timeout',
+  SUPPORT_BUNDLE_EXPIRATION:              'support-bundle-expiration',
+  SUPPORT_BUNDLE_IMAGE:                   'support-bundle-image',
+  SUPPORT_BUNDLE_NODE_COLLECTION_TIMEOUT: 'support-bundle-node-collection-timeout',
```
Comment (Yu-Jack) [inline pkg/harvester/config/settings.js:21]: I only added this one.

File: pkg/harvester/config/settings.js [hunk 2]
```diff
@@ -52,15 +53,16 @@ export const HCI_ALLOWED_SETTINGS = {
...
-  [HCI_SETTING.SUPPORT_BUNDLE_IMAGE]:      { kind: 'json', from: 'import' },
-  [HCI_SETTING.STORAGE_NETWORK]:           { kind: 'custom', from: 'import' },
-  [HCI_SETTING.VM_FORCE_RESET_POLICY]:     { kind: 'json', from: 'import' },
-  [HCI_SETTING.RANCHER_MANAGER_SUPPORT]:   { kind: 'boolean' },
-  [HCI_SETTING.SSL_CERTIFICATES]:          { kind: 'json', from: 'import' },
-  [HCI_SETTING.SSL_PARAMETERS]:            {
+  [HCI_SETTING.OVERCOMMIT_CONFIG]:                      { kind: 'json', from: 'import' },
+  [HCI_SETTING.SUPPORT_BUNDLE_TIMEOUT]:                 {},
+  [HCI_SETTING.SUPPORT_BUNDLE_EXPIRATION]:              {},
+  [HCI_SETTING.SUPPORT_BUNDLE_NODE_COLLECTION_TIMEOUT]: {},
```
Comment (Yu-Jack) [inline pkg/harvester/config/settings.js:59]: And here

**Comment by Yu-Jack** [comment]
This one will also be backported to v1.3.

**Comment by a110605** [comment]
Does `support-bundle-node-collection-timeout` setting allow negative integer or given string ? 
Since `support-bundle-timeout` has webhook validation for negative integer or given string

Screenshot_20240605_125900
Screenshot_20240605_125932
Screenshot_20240605_125950

**Comment by Yu-Jack** [comment]
@a110605 you need to update harvester-webhook image as well.

**Comment by Yu-Jack** [comment]
@a110605 Thanks for testing that part. Already created a PR for that https://github.com/harvester/harvester/pull/5969, and tested it. You could use `jk82421/harvester-webhook:v1.3.0.3` to test it first.

**Comment by a110605** [comment]
> @a110605 Thanks for testing that part. Already created a PR for that [harvester/harvester#5969](https://github.com/harvester/harvester/pull/5969), and tested it. You could use `jk82421/harvester-webhook:v1.3.0.4` to test it first.

Pull your image looks good on my local.

image

**Comment by Yu-Jack** [comment]
@mergify backport release-harvester-v1.3

**Pull request merged**